clients/filesystem-hadoop3/src/test/java/org/apache/gravitino/filesystem/hadoop/integration/test/GravitinoVirtualFileSystemS3CredentialAuthorizationIT.java

```diff
@@ -0,0 +1,343 @@
+        "org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider");
+    properties.put(FILESYSTEM_PROVIDERS, "s3");
+    properties.put(S3Properties.GRAVITINO_S3_REGION, S3_REGION);
+    properties.put(S3Properties.GRAVITINO_S3_ROLE_ARN, S3_ROLE_ARN);
+    properties.put(
+        CredentialConstants.CREDENTIAL_PROVIDERS, S3TokenCredential.S3_TOKEN_CREDENTIAL_TYPE);
+
+    Catalog catalog =
+        metalake.createCatalog(
+            catalogName, Catalog.Type.FILESET, "hadoop", "catalog comment", properties);
+    Assertions.assertTrue(metalake.catalogExists(catalogName));
+    catalog.asSchemas().createSchema(schemaName, "schema comment", properties);
+    Assertions.assertTrue(catalog.asSchemas().schemaExists(schemaName));
+
+    List<SecurableObject> securableObjects = new ArrayList<>();
+    SecurableObject catalogObject =
+        SecurableObjects.ofCatalog(catalogName, ImmutableList.of(Privileges.UseCatalog.allow()));
+    securableObjects.add(catalogObject);
+    SecurableObject schemaObject =
+        SecurableObjects.ofSchema(
+            catalogObject, schemaName, ImmutableList.of(Privileges.UseSchema.allow()));
+    securableObjects.add(schemaObject);
+    metalake.createRole(ROLE_NAME, new HashMap<>(), securableObjects);
+    metalake.grantRolesToUser(ImmutableList.of(ROLE_NAME), NORMAL_USER);
+
+    gvfsConf = new Configuration();
+    gvfsConf.set(
+        "fs.gvfs.impl", "org.apache.gravitino.filesystem.hadoop.GravitinoVirtualFileSystem");
+    gvfsConf.set("fs.AbstractFileSystem.gvfs.impl", "org.apache.gravitino.filesystem.hadoop.Gvfs");
+    gvfsConf.set("fs.gvfs.impl.disable.cache", "true");
+    gvfsConf.set("fs.gravitino.server.uri", serverUri);
+    gvfsConf.set("fs.gravitino.client.metalake", metalakeName);
+    gvfsConf.set("fs.gravitino.enableCredentialVending", "true");
+    gvfsConf.set(S3Properties.GRAVITINO_S3_ENDPOINT, S3_ENDPOINT);
+    gvfsConf.set(S3Properties.GRAVITINO_S3_REGION, S3_REGION);
+    gvfsConf.set(S3Properties.GRAVITINO_S3_ROLE_ARN, S3_ROLE_ARN);
+  }
+
+  @AfterAll
+  public void tearDown() throws IOException {
+    try {
+      if (metalake != null) {
+        Catalog catalog = metalake.loadCatalog(catalogName);
+        catalog.asSchemas().dropSchema(schemaName, true);
+        metalake.dropCatalog(catalogName, true);
+      }
+      if (adminClient != null) {
+        adminClient.dropMetalake(metalakeName, true);
+        adminClient.close();
+        adminClient = null;
+      }
+      if (normalUserClient != null) {
+        normalUserClient.close();
+        normalUserClient = null;
+      }
+      if (client != null) {
+        client.close();
+        client = null;
+      }
+    } finally {
+      try {
+        closer.close();
+      } catch (Exception e) {
+        LOG.error("Exception in closing CloseableGroup", e);
+      }
+    }
+  }
+
+  private Configuration convertGvfsConfigToRealFileSystemConfig(Configuration gvfsConf) {
```

Comment:
how about moving the private method to the bottom?

Reply:
OK.